aea/cli/interact.py

```diff
@@ -149,9 +158,16 @@ def _try_construct_envelope(agent_name: str, sender: str) -> Optional[Envelope]:
             message = message_decoded.encode("utf-8")  # type: Union[str, bytes]
         else:
             message = message_escaped  # pragma: no cover
-        msg = DefaultMessage(performative=performative, content=message)
+        dialogue_reference = dialogues.new_self_initiated_dialogue_reference()
+        msg = DefaultMessage(
+            performative=performative,
+            dialogue_reference=dialogue_reference,
+            content=message,
+        )
         msg.counterparty = agent_name
+        msg.to = agent_name
         msg.sender = sender
+        dialogues.update(msg)
```

Comment:
at the moment the recommended pattern is `dialogue = dialogues.update(msg)` and then `assert dialogue is not None` as otherwise you won't catch any issues during dialogue construction